components/alert/alert.js

```diff
@@ -102,19 +108,24 @@ class Alert extends LocalizeCoreElement(RtlMixin(LitElement)) {
 			.d2l-alert-text-with-actions {
 				padding-right: 0.9rem;
 			}
+			.d2l-alert-text-with-icon {
+				padding-left: 0.5rem;
+			}
 
 			:host([dir="rtl"]) .d2l-alert-text-with-actions {
 				padding-left: 0.9rem;
 				padding-right: 1.5rem;
 			}
-			:host([no-padding]) .d2l-alert-text,
-			:host([no-padding]) .d2l-alert-text-with-actions {
+			:host([no-padding]) .d2l-alert-text {
```

Comment:
With `no-padding`, it is targeting `.d2l-alert-text` _and_ `.d2l-alert-text-with-actions`. If an element belongs to the class `.d2l-alert-text-with-actions`, it also belongs to the class `.d2l-alert-text`:

```
const alertTextClasses = {
    'd2l-alert-text: true,
    'd2l-alert-text-with-actions': <conditional>
};
```

No need for the second half of the query selector.